tests-fuzz/src/generator/create_expr.rs

```diff
@@ -200,6 +203,118 @@ impl<R: Rng + 'static> Generator<CreateTableExpr, R> for CreateTableExprGenerato
+                    options: vec![ColumnOption::TimeIndex],
+                },
+                Column {
+                    name: Ident::new("val"),
+                    column_type: ConcreteDataType::float64_datatype(),
+                    options: vec![],
+                },
+            ],
+            if_not_exists: self.if_not_exists,
+            partition: None,
+            engine: "metric".to_string(),
+            options: [("physical_metric_table".to_string(), "".into())].into(),
+            primary_keys: vec![],
+        })
+    }
+}
+
+/// Generate a logical table based on an existing physical table.
+#[derive(Builder)]
+#[builder(pattern = "owned")]
+pub struct CreateLogicalTableExprGenerator<R: Rng + 'static> {
+    physical_table_ctx: TableContextRef,
+    labels: usize,
+    #[builder(default = "false")]
+    if_not_exists: bool,
+    #[builder(default = "Box::new(WordGenerator)")]
+    name_generator: Box<dyn Random<Ident, R>>,
+    #[builder(default = "Box::new(StringColumnTypeGenerator)")]
+    label_column_type_generator: ConcreteDataTypeGenerator<R>,
+}
+
+impl<R: Rng + 'static> Generator<CreateTableExpr, R> for CreateLogicalTableExprGenerator<R> {
+    type Error = Error;
+
+    fn generate(&self, rng: &mut R) -> Result<CreateTableExpr> {
+        // Currently we mock the usage of GreptimeDB as Prometheus' backend, the physical table must have two columns.
+        ensure!(
+            self.physical_table_ctx.columns.len() == 2,
+            error::UnexpectedSnafu {
+                violated: "The physical table must have two columns"
+            }
+        );
+
+        // Generates the logical table columns based on the physical table.
+        let logical_table_name = self
+            .physical_table_ctx
+            .generate_unique_table_name(rng, self.name_generator.as_ref());
+        let mut logical_table = CreateTableExpr {
+            table_name: logical_table_name,
+            columns: self.physical_table_ctx.columns.clone(),
+            if_not_exists: self.if_not_exists,
+            partition: None,
+            engine: "metric".to_string(),
+            options: [(
+                "on_physical_table".to_string(),
+                self.physical_table_ctx.name.value.clone().into(),
+            )]
+            .into(),
+            primary_keys: vec![],
+        };
+
+        let column_names = self.name_generator.choose(rng, self.labels);
+        logical_table.columns.extend(generate_columns(
+            rng,
+            column_names,
+            self.label_column_type_generator.as_ref(),
+            Box::new(primary_key_column_options_generator),
+        ));
+
+        logical_table.columns.iter_mut().for_each(|column| {
+            if column.column_type == ConcreteDataType::string_datatype() {
+                column
+                    .options
+                    .retain(|option| option == &ColumnOption::PrimaryKey);
+            }
+        });
```

Comment:
```suggestion
```

Reply:
It sounds a little weird but actually `primary_key_column_options_generator` would generate both `primary key` and `not null`. This would fail CI. I actually used it while generating the `__table_id` and `__tsid` line cause the `ColumnOption` struct is private for fuzz test files.

Reply:
Ok, I see. How about only keeping the primary key option? The metric engine will ignore the column's `not null` option.

Reply:
That way the validator would need this piece of code. Maybe we could add another bare primary key generator in ir to avoid this.